packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -371,36 +371,36 @@ export class SeenGossipBlockInput {
 
   private async reconstructColumns(config: ChainForkConfig, blockRoot: RootHex) {
     const blockCache = this.blockInputCache.get(blockRoot);
-    if (blockCache === undefined || blockCache.fork !== ForkName.fulu) {
+    if (
+      blockCache === undefined ||
+      blockCache.fork !== ForkName.fulu ||
+      blockCache.cachedData === undefined ||
+      blockCache.cachedData.fork !== ForkName.fulu
```

Comment:
I see that you combined these from the other conditional below but they don't look like they are doing the intended check.  Please double check this is what you originally intended

Reply:
I'll roll all these updated checks back. I did this because `getEmptyBlockInputCacheEntry` always returns `cachedData` for Fulu, but if the types don't assert this, then I agree it's not worth risking a bug